packages/dds/tree/src/test/shared-tree/schematizingTreeView.spec.ts

```diff
@@ -300,5 +300,39 @@ describe("SchematizingSimpleTreeView", () => {
 			assert.equal(localChanges, 1);
 			unsubscribe();
 		});
+
+		it("does not emit changed events for rebases", () => {
+			const stringArraySchema = schema.array([schema.string]);
+			const stringArrayStoredSchema = toStoredSchema(stringArraySchema);
+			const stringArrayContent = {
+				schema: stringArrayStoredSchema,
+				initialTree: cursorFromInsertable(stringArraySchema, ["a", "b", "c"]),
+			};
+			const checkout = checkoutWithContent(stringArrayContent);
+			const main = new SchematizingSimpleTreeView(
+				checkout,
+				new TreeViewConfiguration({ schema: stringArraySchema }),
+				new MockNodeKeyManager(),
+			);
+			const branch = main.fork();
+
+			let changes = 0;
+
+			const unsubscribe = branch.events.on("changed", (data) => {
+				changes++;
+			});
+
+			const mainRoot = main.root;
+			const branchRoot = branch.root;
+
+			branchRoot.removeAt(1);
```

Comment:
Consider simplifying the edits here to be the minimal thing we need:

1. A single edit to the main branch
2. A rebase of the branch over the main branch

It's important to have (1) because we want to make sure we don't hit a "no-op rebase" fast path (which might skip emitting an event, even if it in general did - incorrectly - emit an event). But that's all we require here.